File: pdu/python/geometry_msgs/pdu_cdr_conv_TwistStamped.py
from .pdu_pytype_TwistStamped import TwistStamped
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *
from ..geometry_msgs.pdu_cdr_conv_Twist import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_TwistStamped(writer: CdrWriter, src: TwistStamped):
    py_to_cdr_body_Header(writer, src.header)
    py_to_cdr_body_Twist(writer, src.twist)


def cdr_body_to_py_TwistStamped(reader: CdrReader, dst: TwistStamped):
    cdr_body_to_py_Header(reader, dst.header)
    cdr_body_to_py_Twist(reader, dst.twist)
    return dst


def py_to_cdr_TwistStamped(src: TwistStamped) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_TwistStamped(writer, src)
    return writer.bytes()


def cdr_to_py_TwistStamped(cdr_payload) -> TwistStamped:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = TwistStamped()
    return cdr_body_to_py_TwistStamped(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_TwistStamped.js
import { TwistStamped } from './pdu_jstype_TwistStamped.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';
import { Twist } from '../geometry_msgs/pdu_jstype_Twist.js';
import { PduTwistConverter } from '../geometry_msgs/pdu_cdr_conv_Twist.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduTwistStampedConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { TwistStamped } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        PduTwistConverter.to_cdr_body(writer, src.twist);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { TwistStamped } dst
     * @returns { TwistStamped }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduTwistConverter.cdr_body_to_js(reader, dst.twist);
        return dst;
    }

    /**
     * @param { TwistStamped } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { TwistStamped }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new TwistStamped());
    }
}
